--- ConfigDefs.pl	2024-06-12 17:11:05.587164565 -0400
+++ ConfigDefs.pl.new	2024-06-12 17:10:49.022863933 -0400
@@ -264,6 +264,7 @@
 workuser			= incomingworkuser
 workgroup			= incomingworkgroup
 workperms			= incomingworkpermissions
+eximcommand			= EximCommand
 
 
 #
@@ -697,4 +698,5 @@
 usespamcache            0       no      0       yes     1
 spamcachedatabasefile
 spamcachetimings
+EximCommand             /opt/exim4/bin/exim
 ## end MailCleaner
