--- MailScanner	2024-06-17 16:36:51.033275535 -0400
+++ MailScanner.new	2024-06-17 17:14:52.554514979 -0400
@@ -377,6 +377,36 @@
 $_=MailScanner::Config::QuickPeek($ConfFile, 'mta');
 $_='sendmail' if $WantLintOnly || $WantLintLiteOnly || $WantRuleCheck;
 if (/exim/i) {
+  # Check for version >= 4.97
+  my $eximcommand = MailScanner::Config::QuickPeek($ConfFile, 'eximcommand');
+  if ($eximcommand eq "") {
+    # try to find exim
+    eval {
+      $eximcommand = `which exim`;
+    };
+    if (defined($@) && $@ ne '') {
+      print STDERR "Exim binary not found. Please supply the full path to Exim Command in MailScanner.conf";
+      exit 1;
+    }
+  }
+  # Get exim version
+  my @out;
+  my $ver;
+  eval {
+    @out = `$eximcommand -bV`;
+    my @line = split(/ /, $out[0]);
+    $ver = $line[2];
+    $ver =~ /\d+\.\d+/;
+  };
+  if ($@) {
+    print STDERR "Unable to obtain version of Exim. Please check the path to Exim Command in MailScanner.conf";
+    exit 1;
+  }
+  if ($ver >= 4.97) {
+    MailScanner::Config::SetValue('eximlongids',1);
+  } else {
+    MailScanner::Config::SetValue('eximlongids',0);
+  }
   $MTAmod = 'Exim.pm';
   $MTADSmod = 'EximDiskStore.pm';
 } elsif(/zmailer/i) {
