--- Message.pm	2021-04-06 14:56:17.783103144 -0400
+++ Message.pm.new	2021-04-06 14:51:09.242974313 -0400
@@ -7234,13 +7234,14 @@
   return @DisarmDoneSomething;
 }
 
+our ($DisarmInsideHead, $DisarmInsideScript) = ( 0, 0 );
+
 # HTML::Parser callback for text so we can collect the contents of links
 sub DisarmTextCallback {
   my($text) = @_;
 
-  unless ($DisarmInsideLink) {
+  unless ($DisarmInsideLink || ( $DisarmInsideHead && $DisarmInsideScript )) {
     print $text;
-    #print STDERR "DisarmText just printed \"$text\"\n";
     return;
   }
 
@@ -7250,8 +7251,6 @@
   #print STDERR "DisarmText just added \"$text\"\n";
 }
 
-our $DisarmInsideHead = 0;
-
 # HTML::Parser callback function for start tags
 sub DisarmTagCallback {
   my($tagname, $text, $attr, $attrseq) = @_;
@@ -7263,6 +7262,7 @@
 
   if ($tagname eq 'head') {
     $DisarmInsideHead = 1;
+    $output .= $text;
     #MailScanner::Log::InfoLog("Entering head\n");
   } elsif ($tagname eq 'form' && $DisarmFormTag) {
     #print "It's a form\n";
@@ -7310,6 +7310,7 @@
   } elsif ($tagname eq 'script' && $DisarmScriptTag) {
     #print "It's a script\n";
     $text = substr $text, 1;
+    $DisarmInsideScript = 1;
     unless ($DisarmInsideHead) {
       $output .= "<MailScannerScript$$ style=\"display: none;\" " . $text;
     }
@@ -7404,6 +7405,7 @@
 
   if ($tagname eq 'head') {
     $DisarmInsideHead = 0;
+    print $text;
     #MailScanner::Log::InfoLog("Exitting head\n");
   } elsif ($tagname eq 'iframe' && $DisarmIframeTag) {
     print "</MailScannerIFrame$$>";
@@ -7412,6 +7414,7 @@
     print "</MailScannerForm$$>";
     $DisarmDoneSomething{'form'} = 1;
   } elsif ($tagname eq 'script' && $DisarmScriptTag) {
+    $DisarmInsideScript = 0;
     unless ($DisarmInsideHead) {
       print "</MailScannerScript$$>";
     }
